src/main/java/com/atlassian/db/replica/internal/SqlQuery.java

```diff
@@ -27,4 +43,12 @@ private boolean isUpdate() {
     private boolean isDelete() {
         return sql != null && (sql.startsWith("delete") || sql.startsWith("DELETE"));
     }
+
+    private String trimForSelectForUpdateCheck() {
+        if (sql.length() < SELECT_FOR_UPDATE_SUFFIX_LIMIT) {
+            return sql;
+        } else {
+            return sql.substring(sql.length() - SELECT_FOR_UPDATE_SUFFIX_LIMIT);
```

Comment:
considering that they still will "SELECT FOR UPDATES" hidden in 100+ chars, (very low likelihood) how they can be noticed?